### ISSUE TITLE
Missing invalid syntax in ES7/2016

### ISSUE THREAD

**Issue by nzakas** [eslint/espree#284]
```
function foo(a = 1) {
  "use strict"
}
```

In ES2016, this code raises a syntax error.

> It is a Syntax Error if ContainsUseStrict of FunctionBody is true and IsSimpleParameterList of FormalParameters is false.
https://tc39.github.io/ecma262/2016/#sec-function-definitions-static-semantics-early-errors

However, Espree parses this without problem. It should throw an error only when `ecmaVersion` is greater than 6.

(Note: fixed upstream in Acorn, we just have to merge here.)
